estate/models/estate_property.py

```diff
@@ -0,0 +1,116 @@
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Advertisement module"
+    _order = "id desc"
+
+    name = fields.Char('Name', required=True)
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Date availability', copy=False, default=default_availability_date)
+    expected_price = fields.Float('Expected price', required=True)
+    selling_price = fields.Float('Selling price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area')
+    garden_orientation = fields.Selection(selection=[
+        ('north', 'North'),
+        ('south', 'South'),
+        ('east', 'East'),
+        ('west', 'West')
+    ], string='Garden orientation')
+    total_area = fields.Integer('Total Area', compute='_compute_total_area')
+    active = fields.Boolean('Active', default=True)
+    state = fields.Selection(selection=[
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')
+    ], string="State", default='new', required=True, copy=False)
+    property_type_id = fields.Many2one('estate.property.type', string="Property type")
+    property_buyer_id = fields.Many2one('res.partner', string='Buyer', copy=False)
+    property_salesperson_id = fields.Many2one('res.users', string="Salesperson", default=lambda self: self.env.user)
+    property_tag_ids = fields.Many2many('estate.property.tag', string="Property tags")
+    property_offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    best_price = fields.Float('Best Price', compute='_compute_best_price')
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)',
+        'The expected price must be strictly positive.'
+    )
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price >= 0)',
+        'The selling price must be positive.'
+    )
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('property_offer_ids.price')
+    def _compute_best_price(self):
+        for record in self:
+            offer_prices = record.property_offer_ids.mapped('price')
+            record.best_price = max(offer_prices) if offer_prices else 0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
```

Comment:
Use `False` (or `None`) to clear a selection field. Using an empty string `''` works but isn't standard practice in Odoo.
```suggestion
            self.garden_orientation = False
```